mycroft/skills/skill_updater.py

```diff
@@ -13,28 +13,84 @@
 # limitations under the License.
 #
 """Periodically run by skill manager to update skills and post the manifest."""
+import json
 import os
-import sys
-from datetime import datetime
-from time import time
+from os.path import join, isfile, isdir
 import xdg.BaseDirectory
+from json_database import JsonStorage
 
 from mycroft.api import DeviceApi, is_paired
 from mycroft.configuration import Configuration
 from ovos_utils.configuration import get_xdg_base, is_using_xdg
 from mycroft.util import connected
 from combo_lock import ComboLock
 from mycroft.util.log import LOG
-from mycroft.skills.msm_wrapper import build_msm_config, create_msm, MsmException
 from mycroft.util.file_utils import get_temp_path
+from mycroft.skills.skill_loader import get_skill_directories
+
 
 ONE_HOUR = 3600
 FIVE_MINUTES = 300  # number of seconds in a minute
 
 
 def skill_is_blacklisted(skill):
-    blacklist = Configuration.get()['skills']['blacklisted_skills']
-    return os.path.basename(skill.path) in blacklist or skill.name in blacklist
+    # this is a internal msm helper
+    # it should have been private
+    # cant remove to keep api compatibility
+    # nothing in the wild should be using this
+    LOG.warning("skill_is_blacklisted is an internal method and has been deprecated. Stop using it!")
+    return False
+
+
+class SeleneDeviceSkills(JsonStorage):
+    """dict subclass with save/load support
+    This dictionary contains the metadata expected by selene backend
+    This data is used to populate entries in selene skill settings page
+    It is only uploaded if enabled in mycroft.conf and device is paired
+    """
+    def __init__(self, api=None):
+        path = os.path.join(xdg.BaseDirectory.save_data_path(get_xdg_base()), 'skills.json')
+        super().__init__(path)
+        if "skills" not in self:
+            self["skills"] = {}
+        self.api = api or DeviceApi()
+
+    def device_skill_state_hash(self):
+        return hash(json.dumps(self, sort_keys=True))
+
+    def add_skill(self, skill_id):
+        if self.api.identity.uuid:
+            skill_gid = f'@{self.api.identity.uuid}|{skill_id}'
+        else:
+            skill_gid = f'@|{skill_id}'
+        skill = {
+            "name": skill_id,
+            "origin": "non-msm",
+            "beta": True,
+            "status": 'active',
+            "installed": 0,
+            "updated": 0,
+            "installation": 'installed',
+            "skill_gid": skill_gid
+        }
+        self["skills"].append(skill)
+
+    def get_skill_state(self, skill_id):
+        """Find a skill entry in the device skill state and returns it."""
+        for skill_state in self.get('skills', []):
```

Comment:
init defines this as a dict, but default value here is a list?

Reply:
if undefined we just want to skip the loop, i used a list because i didnt look up what the contents actually were

Reply:
Fair enough